cmd/atrutil/atrutil_main.go

```diff
@@ -220,6 +222,140 @@ func FillInToolPathDefaults() {
+	var atomicTests = map[string][]*types.TestSpec{} // tid -> tests
+
+	errRead := utils.LoadAtomicsIndexCsv(filepath.FromSlash(flagAtomicsPath), &atomicTests)
+	if errRead != nil {
+		fmt.Println("Unable to load Indexes-CSV file for Atomics", errRead)
+		return errRead, 0.0
+	}
+
+	percentage := FindTestCoverageHelper(flagCriteriaPath, &atomicTests)
+
+	return nil, percentage
+}
+
+func FindTestCoverageHelper(dirPath string, atomicMap *map[string][]*types.TestSpec) float32 {
+	dirPath = filepath.FromSlash(dirPath)
+	allfiles, err := ioutil.ReadDir(dirPath)
+
+	platformName := utils.GetPlatformName()
+	var percentage float32 = 0.0
+
+	readErr := utils.LoadAtomicsIndexCsv(filepath.FromSlash(flagAtomicsPath), atomicMap)
+	if readErr != nil {
+		fmt.Println("Unable to load Indexes-CSV file for Atomics", err)
+		os.Exit(1)
+	}
+	total := 0
+
+	criteria := 0
+	for _, entries := range *atomicMap {
+		for range entries {
+			total += 1
+		}
+	}
+	fmt.Println("Found", total, "tests for platform", platformName)
+
+	if err != nil {
+		fmt.Println("ERROR: unable to list files in "+dirPath, err)
+		return 0.0
+	}
+	for _, f := range allfiles {
+		if !strings.HasSuffix(f.Name(), ".csv") {
+			continue
+		}
+
+		if strings.Contains(f.Name(), "_withguids") {
+			continue
+		}
+
+		if gVerbose {
+			fmt.Println("Loading " + f.Name())
+		}
+
+		criteria += FindCoverage(filepath.FromSlash(dirPath+"/"+f.Name()), *atomicMap)
+		if err != nil {
+			fmt.Println("ERROR:", err)
+			return 0.0
+		}
+	}
+
+	percentage = float32(criteria) / float32(total)
+
+	fmt.Printf("Percent coverage for %s: %f%%", platformName, percentage)
+
+	return percentage
+}
+
+func FindCoverage(filename string, atomicMap map[string][]*types.TestSpec) int {
+	platformName := utils.GetPlatformName()
+
+	fmt.Printf("finding coverage for %s for platform %s\n", filename, platformName)
+	filename = filepath.FromSlash(filename)
+
+	infile, err := os.OpenFile(filename, os.O_RDONLY, 0644)
+	if err != nil {
+		fmt.Println("ERROR: unable to open file", filename, err)
+		os.Exit(2)
+	}
+
+	defer infile.Close()
+
+	scanner := bufio.NewScanner(infile)
+
+	//define local variable to count number of occurences of criteria for particular tests
+	criteria := 0
+
+	for scanner.Scan() {
+		line := scanner.Text()
+		fmt.Print(line, "\n")
+
+		r := csv.NewReader(bytes.NewReader([]byte(line)))
+		r.LazyQuotes = true
+		r.Comment = '#'
+		r.FieldsPerRecord = -1 // no validation on num columns per row
+
+		row, err := r.Read()
+
+		if err == nil && strings.HasPrefix(row[0], "T") {
+
+			cur := utils.AtomicTestCriteriaNew(row[0], row[1], row[2], row[3])
+
+			for _, entry := range atomicMap[cur.Technique] {
+				if len(cur.TestGuid) > 0 && strings.HasPrefix(entry.TestGuid, cur.TestGuid) {
+					criteria += 1
+					break
+				}
+				fmt.Print("Current Test Index: ", cur.TestIndex, "\nEntry TestIndex: ", entry.TestIndex, "\n")
+				fmt.Print("Current Test Name: \n", cur.TestName, "\n")
```

Comment:
you probably want to hide these debug logging statements in a `if gVerbose {}` block